src/shared/parse-helpers.c

```diff
@@ -196,3 +198,23 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int parse_open_file_fields(const char *v, char **path, char **fdname, int *flags) {
+    _cleanup_strv_free_ char **parts = NULL;
+    int r;
+
+    if (!v)
+        return -EINVAL;
+
+    r = strv_split_full(&parts, v, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+    if (r < 0)
+            return r;
+    if (strv_length(parts) != 3)
+            return -EINVAL;
+
+    *path = strdup(parts[0]);
+    *fdname = strdup(parts[1]);
+    *flags = streq(parts[2], "rw") ? O_RDWR : O_RDONLY;
```

Comment:
We should define a separate set of bitflags specifically for the `OpenFile=` setting and translate those to the corresponding `open()` flags just before we call `open()`. Since we might add more flags in the future that can't be represented by an `open()` flag. See `enum PortableFlags` in portable.h for an example of a public flags enum.